### PR TITLE
fix(kradio): update radio card styles [KHCP-9491]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

[Jira](https://konghq.atlassian.net/browse/KHCP-9570)
[Figma](https://www.figma.com/file/Yze0SWXl5nKjR0rFdilljK/Components?type=design&node-id=763%3A12969&mode=dev)

Updates KRadio card styles

Screen Shot 2023-11-16 at 5 19 48 PM

Links:
- [Kongponents sandbox](https://deploy-preview-1851--kongponents-sandbox.netlify.app/#/radio)
- [Kongponents docs preview](https://deploy-preview-1851--kongponents.netlify.app/components/radio.html)

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

File: src/components/KRadio/KRadio.vue [hunk 1]
```diff
@@ -351,23 +344,24 @@ $kRadioDotSize: 6px;
     .radio-card-wrapper {
       align-items: center;
       background-color: var(--kui-color-background, $kui-color-background);
+      border: var(--kui-border-width-10, $kui-border-width-10) solid var(--kui-color-border, $kui-color-border);
```
Comment (adamdehaven) [inline src/components/KRadio/KRadio.vue:347]: I think we need to use a box-shadow instead of border for the selected cards. If no cards are selected, and then you select one, the cards "jump" on the page in their container when the border is added. Alternatively, you could give all cards the same border-width border, but `transparent` by default, and then add the color when the item is selected.

File: src/components/KRadio/KRadio.vue [hunk 2]
```diff
@@ -376,51 +370,83 @@ $kRadioDotSize: 6px;
 
       &.has-label, &.has-description {
         .card-content-wrapper {
+          height: auto;
```
Comment (adamdehaven) [inline src/components/KRadio/KRadio.vue:373]: nit/question: why is this needed? Isn't it the default?

File: src/components/KRadio/KRadio.vue [hunk 3]
```diff
@@ -376,51 +370,83 @@ $kRadioDotSize: 6px;
...
         }
       }
 
       .radio-label {
         @include labelDefaults;
       }
+
+      :deep(#{$kongponentsKongIconSelector}) {
+        color: var(--kui-color-text-neutral, $kui-color-text-neutral) !important;
+      }
```
Comment (adamdehaven) [inline src/components/KRadio/KRadio.vue:384]: issue: Shouldn't the host app have control over the icon it slots? There would be no way to override this.

File: src/components/KRadio/KRadio.vue [hunk 4]
```diff
@@ -376,51 +370,83 @@ $kRadioDotSize: 6px;
...
+          border: var(--kui-border-width-20, $kui-border-width-20) solid var(--kui-color-border-primary-strong, $kui-color-border-primary-strong);
+        }
+
+        .radio-label {
+          color: var(--kui-color-text-primary, $kui-color-text-primary);
+        }
+
+        :deep(#{$kongponentsKongIconSelector}) {
+          color: var(--kui-color-text-primary, $kui-color-text-primary) !important;
+        }
```
Comment (adamdehaven) [inline src/components/KRadio/KRadio.vue:401]: issue: Shouldn't the host app have control over the icon it slots? There would be no way to override this.

File: src/components/KRadio/KRadio.vue [hunk 5]
```diff
@@ -376,51 +370,83 @@ $kRadioDotSize: 6px;
...
+            border: var(--kui-border-width-20, $kui-border-width-20) solid var(--kui-color-border-danger-strong, $kui-color-border-danger-strong);
+          }
+
+          .radio-label {
+            color: var(--kui-color-text, $kui-color-text);
+          }
+
+          :deep(#{$kongponentsKongIconSelector}) {
+            color: var(--kui-color-text, $kui-color-text) !important;
+          }
```
Comment (adamdehaven) [inline src/components/KRadio/KRadio.vue:432]: issue: Shouldn't the host app have control over the icon it slots? There would be no way to override this.

File: src/components/KRadio/KRadio.vue (hunk 1 above)
Comment (portikM) [reply src/components/KRadio/KRadio.vue:347]: Fixed

File: src/components/KRadio/KRadio.vue (hunk 2 above)
Comment (portikM) [reply src/components/KRadio/KRadio.vue:373]: This is needed, it overrides another rule just 3 lines above it

File: src/components/KRadio/KRadio.vue (hunk 3 above)
Comment (portikM) [reply src/components/KRadio/KRadio.vue:384]: Figma designs define icon color in different states so this should be controlled by the component.

File: src/components/KRadio/KRadio.vue (hunk 4 above)
Comment (portikM) [reply src/components/KRadio/KRadio.vue:401]: Same as above, Figma designs define icon color in different states so this should be controlled by the component.

File: src/components/KRadio/KRadio.vue (hunk 5 above)
Comment (portikM) [reply src/components/KRadio/KRadio.vue:432]: Same as above.

File: src/components/KRadio/KRadio.vue (hunk 3 above)
Comment (adamdehaven) [reply src/components/KRadio/KRadio.vue:384]: This means that we can NEVER use another color icon within the component? Is that desirable? What if we're using the "kong" icon with gradient, or one of the flag icons down the road?

Comment (portikM) [reply src/components/KRadio/KRadio.vue:384]: It can't be overwritten using `id` selector.
```
...
<KongIcon id="specific-color-icon" />
...
#specific-color-icon {
  color: $kui-color-text-success !important;
}
```

Screen Shot 2023-11-20 at 10 30 28 AM

Comment (adamdehaven) [reply src/components/KRadio/KRadio.vue:384]: Doesn't seem ideal IMO; that's the whole point of the `color` prop on the icons?

Comment (portikM) [reply src/components/KRadio/KRadio.vue]: Fixed now
